app/controllers/concerns/hyrax/controller.rb

```diff
@@ -52,7 +52,7 @@ def set_locale
   # render a json response for +response_type+
   def render_json_response(response_type: :success, message: nil, options: {})
     json_body = Hyrax::API.generate_response_body(response_type: response_type, message: message, options: options)
-    render json: json_body, status: response_type
+    render json: json_body, status: Hyrax::API::DEFAULT_RESPONSES[response_type][:code]
```

Comment:
This could easily itself raise an error (e.g. the `response_type` is missing in the map).  I think a `Hyrax::API.response_code_for(response_type)` method would help insulate from a failure that could be avoidable.

Reply:
I agree this could be cleaned up.  I created this WIP PR so I wouldn't forget as I was working through something else.  Failing test would also be good to add.  I'll come back to this eventually.